### PR TITLE
Second POST call for creating Migration Request

### PR DESCRIPTION
**Pull request by AparnaKarve**
Both API calls for -
- Creating a Migration Plan
- Creating a Migration Request

have been implemented in the `mockMode` as well as real API mode.

(Requires testing in real API mode)

### CODE REVIEW

**Comment by AparnaKarve** [comment]
The `if (modeMode)`/`else` block can be refactored if necessary.

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js [hunk 1]
```diff
@@ -1,19 +1,72 @@
...
-  dispatch({
-    type: POST_V2V_MIGRATION_PLANS,
-    payload: API.post(url, migrationPlans)
-  }).catch(error => {
-    // to enable UI development without the backend ready, i'm catching the error
-    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
+const mockMode = false;
+
+const _postMigrationPlansActionCreator = (url, migrationPlans) => dispatch => {
+  if (mockMode) {
```
Comment (jeff-phillips-18) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js]: in mockMode should we bother calling the API or just go straight to dispatching the mock data?

Comment (AparnaKarve) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js]: I have been mostly following the existing pattern in other Action creators - although I agree, we could skip calling the API.

**Comment by AparnaKarve** [comment]
I have tested the 1st POST successfully.

As for the second POST, I'm getting a 500.
One of the methods in the Rails code seems to be blowing up - will discuss this with Bill tomorrow.

**Comment by mturley** [comment]
@AparnaKarve looks like you have an eslint failure:

```
/home/travis/build/priley86/miq_v2v_ui_plugin/app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js
  50:24  error  Promise constructor parameters must be named resolve, reject  promise/param-names
```

If we're having issues with a 500 on the real API, can we put this in `mockMode` and get it merged? then another PR to turn off and test without `mockMode`?

**Comment by AparnaKarve** [comment]
@mturley Yes, currently looking into that error (that I don't agree with)
We may have to change some linting rules if necessary for this one.
Will let you know.

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js (hunk 1 above)
Comment (jeff-phillips-18) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardResultsStep/PlanWizardResultsStepActions.js]: That's fine. I'd like to see an overall mock flag as well to be able to turn them all on from one place at some point. We can work that all out later.

**Comment by AparnaKarve** [comment]
eslint errors have been addressed

Other updates: 
- The 500 error from the second POST is now resolved.
- successfully conducted end-to-end testing in real API mode

**Comment by AparnaKarve** [comment]
NOTE that for the POC, the .csv in step2, needs to have VM id's in the `reference` column.

In real API mode, these VM id's need to actually exist in your DB.

A reference CSV for the POC is as shown below -
```csv
bostonVM1, prov1, 1 
raleighVM2, prov2, 2 
```

**Comment by mturley** [comment]
Was about to approve, you beat me to it :) Great job on this @AparnaKarve .

**Pull request merged**